app/scripts/lib/WeakRefObjectMap.ts

```diff
@@ -18,13 +18,13 @@ type WeakRefCompatibleObject<T> = {
  * referenced so that they can be garbage collected if/when a dapp connection ends without effective cleanup.
  */
 
-export class WeakRefObjectMap<WeakReffedObject extends Record<string, object>>
-  implements Map<string, WeakReffedObject>
+export class WeakRefObjectMap<T extends Record<string, WeakKey>>
```

Comment:
not that is actually different in this case but pretty sure this can just be
```suggestion
export class WeakRefObjectMap<T extends Record<string, object>>
```